### PR TITLE
Learning Walk API

### PR DESCRIPTION
**Pull request by rishipisipati**
Creating the Learning Walk Microservice

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,201 @@
+plugins {
+    id 'org.springframework.boot' version '2.5.6'
+    id 'io.spring.dependency-management' version '1.0.11.RELEASE'
+    id 'org.asciidoctor.convert' version '1.5.8'
+    id 'java'
+    id 'jacoco'
+    // id 'checkstyle'
```
Comment (bl-jamesli) [inline build.gradle:7]: I really want to have `checkstyle` standardized everywhere if possible.

File: src/main/resources/application-docker.properties [hunk 2]
```diff
@@ -0,0 +1,15 @@
+spring.datasource.url=jdbc:postgresql://db:5432/learningwalk_api?stringtype=unspecified
```
Comment (bl-jamesli) [inline src/main/resources/application-docker.properties:1]: probably define a different postgres port for docker, just like you did for local?

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+# disable startup banner
+spring.main.banner-mode=off
+
+# tweaks to make it runs faster in containers
+spring.jmx.enabled=false
+
+version.hash=${GIT_COMMIT_HASH:0000000000}
+version.timestamp=${GIT_COMMIT_TIMESTAMP:0000000000}
+
+me.client-id=survey-service-test
```
Comment (bl-jamesli) [inline src/main/resources/application.properties:32]: just want to make sure do you really mean survey service, or should it be learning walk related stuff

**Review by bl-jamesli — COMMENTED** [review]
some nits here and there

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+    implementation 'com.google.code.gson:gson:2.8.9'
+    implementation 'com.google.code.findbugs:jsr305:3.0.2'
+    implementation 'com.google.guava:guava:31.0.1-jre'
+    implementation 'io.jsonwebtoken:jjwt:0.9.1'
+    implementation 'com.nimbusds:nimbus-jose-jwt:8.3'
+    implementation 'org.bouncycastle:bcprov-jdk15on:1.61'
+    implementation 'org.bitbucket.b_c:jose4j:0.3.7'
+    implementation 'junit:junit:4.13.1'
+    implementation 'com.sendgrid:sendgrid-java:4.8.3'
```
[the remark this answers is not in the document]
Comment (rishipisipati) [reply build.gradle:56]: For now, I have copied the dependencies from survey service. As and when dependencies aren't needed, they will be cleared out in the future.

**Review by bl-jamesli — APPROVED** [review]

File: src/main/resources/application.properties (hunk 3 above)
Comment (rishipisipati) [reply src/main/resources/application.properties]: You are right, this is incorrect.

File: src/main/resources/application-docker.properties (hunk 2 above)
Comment (rishipisipati) [reply src/main/resources/application-docker.properties:1]: Not working for now, but this isn't a blocker for this story.

**Pull request merged**